runtime/cudaq/algorithms/get_state.h

```diff
@@ -118,6 +119,17 @@ auto get_state(QuantumKernel &&kernel, Args &&...args) {
     return state(new RemoteSimulationState(std::forward<QuantumKernel>(kernel),
                                            std::forward<Args>(args)...));
   }
+#endif
+#if defined(CUDAQ_QUANTUM_DEVICE)
+  // Store kernel name and arguments for quantum states.
+  if (!cudaq::get_quake_by_name(cudaq::getKernelName(kernel), false).empty()) {
+    return state(new QuantumState(std::forward<QuantumKernel>(kernel),
+                                  std::forward<Args>(args)...));
+  } else {
+    throw std::runtime_error(
+        "cudaq::state* argument synthesis is not supported for quantum hardware"
+        "for c-like functions, use class kernels instead");
+  }
 #endif
```

Comment:
```suggestion
#else
```
and wrap the return in the else conditional block...